### PR TITLE
 refactor: drop GetPodByVirtualMachineInstance,getPodsByLabel . Use `libpod.GetPodByVirtualMachineInstance` instead at all its callers.

### LINKED ISSUE #11440: tests/utils.go: Drop GetPodByVirtualMachineInstance from utils.go and replace it in all callers to libpod.GetPodByVirtualMachineInstance
**Is your feature request related to a problem? Please describe**:

The functions `getPodsByLabel` and `GetPodByVirtualMachineInstance` in tests/utils.go should be dropped and replaced by calls to `libpod.GetPodByVirtualMachineInstance`. Also since this is test code, assertions should be used instead of panicking on errors.

**Describe the solution you'd like**:

Drop functions `getPodsByLabel` and `GetPodByVirtualMachineInstance` in tests/utils.go and replace calls to `GetPodByVirtualMachineInstance` with `libpod.GetPodByVirtualMachineInstance`.

### PR DESCRIPTION
**Pull request by TheRealSibasishBehera**
### What this PR does
Before this PR:

The function `GetPodByVirtualMachineInstance` in `tests/utils.go` was used to get pods using labels. 

After this PR:

- Drop `GetPodByVirtualMachineInstance`  and `getPodsByLabel `from `utils.go`.
- Instead used `libpod.GetPodByVirtualMachineInstance` for its callers

Fixes #11440

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by EdDev — CHANGES_REQUESTED** [review]
The original issue this PR treats needs to be changed.

Please drop `GetPodByVirtualMachineInstance` from `utils.go` and replace it in all callers to `libpod.GetPodByVirtualMachineInstance` that already exists.

File: tests/utils.go [hunk 1]
```diff
@@ -227,17 +227,6 @@ func getRunningPodByVirtualMachineInstance(vmi *v1.VirtualMachineInstance, names
...
-	util2.PanicOnError(err)
-
-	if len(pods.Items) != 1 {
-		util2.PanicOnError(fmt.Errorf("found wrong number of pods for VMI '%v', count: %d", vmi, len(pods.Items)))
-	}
-
-	return &pods.Items[0]
-}
-
 func getPodsByLabel(label, labelType, namespace string) (*k8sv1.PodList, error) {
```
Comment (EdDev) [inline tests/utils.go:230]: Is it used now?

Comment (TheRealSibasishBehera) [reply tests/utils.go:230]: no it's unused now
should I remove this as well?

Comment (EdDev) [reply tests/utils.go:230]: Yes

**Review by EdDev — COMMENTED** [review]
Thank you.

Last thing:
- The commit message line is too long, consider:
```
 refactor: drop GetPodByVirtualMachineInstance,getPodsByLabel
 
 Use `libpod.GetPodByVirtualMachineInstance` instead at all its callers.
```
- PR subject: Please use the same as the commit.

Other than that, LGTM.

**Review by EdDev — APPROVED** [review]
Thank you.

/test all

**Review by 0xFelix — APPROVED** [review]
Thanks, looks good!

@EdDev Thanks for taking care of my mixup.

**Review by enp0s3 — COMMENTED** [review]
/approve

**Pull request merged**